sphinx/directives/code.py

```diff
@@ -156,6 +154,190 @@ def run(self):
         return [literal]
 
 
+class LiteralIncludeReader(object):
+    INVALID_OPTIONS_PAIR = [
+        ('pyobject', 'lines'),
+        ('lineno-match', 'lineno-start'),
+        ('lineno-match', 'append'),
+        ('lineno-match', 'prepend'),
+        ('start-after', 'start-at'),
+        ('end-before', 'end-at'),
+        ('diff', 'pyobject'),
```

Comment:
I added new validation for combination of `diff` and other options. I need to confirm this is correct or not. But it seems some cases are broken at this moment.